### ISSUE TITLE
How to define a task-specific validation measure?

### ISSUE THREAD

**Issue by amirj** [kyunghyuncho/dl4mt-material#33]
In the current implementation, validError is computed according to log probabilities ('pred_probs' function). Is it possible to define a **task specific validation error**?
Do you recommend it?
please explain which part of the source code should be modified.
Thanks.

**Comment by orhanf** [comment]
@amirj what do you mean by **task specific validation error**, something like BLEU score may be? But in any case you should just need to change the internals of `pred_probs` function.

**Comment by amirj** [comment]
@orhanf Yes, BLEU or any other validation measures in sequence to sequence learning. I think it should be customizable if possible.

**Comment by amirj** [comment]
@orhanf what about **training loss function**?
Is it rational to change it to another function? or the loss function should pass some constraints?

**Comment by orhanf** [comment]
@amirj i think this is a good research question, to my knowledge there are some efforts towards that, like [task loss estimation](http://arxiv.org/pdf/1511.06456.pdf) and [minimum risk training](http://arxiv.org/pdf/1512.02433.pdf).

**Comment by amirj** [comment]
@orhanf  I just interested to know if 'nmt.py' has a customizable function for computing loss over training data?

**Comment by orhanf** [comment]
I see, change [this](https://github.com/kyunghyuncho/dl4mt-material/blob/master/session2/nmt.py#L652-L657) section accordingly then.

**Comment by amirj** [comment]
@orhanf Kindly appreciate your collaboration. Thank you.
